Resources/doc/advanced-usage-with-pagerfanta-bundle.md

```diff
@@ -0,0 +1,275 @@
+            }
+        } catch (\RuntimeException $e) {
+            $sorting = [];
+        }
+
+        if (null === $form) {
+            $queryBuilder = $qb ?? $this->getCollectionQueryBuilder();
+
+            if (!empty($sorting)) {
+                $this->applySorting($queryBuilder, $sorting);
+            }
+
+            return $this->paginate($options, $queryBuilder);
+        }
+
+        return $this->paginate(
+            $options,
+            $this->getFilteredResource($form, $sorting, $qb),
+        );
+    }
+    
+    public function getPager(QueryBuilder $qb, int $page = 1, int $maxPerPage = 10): Pagerfanta
+    {
+        $pager = new Pagerfanta(new QueryAdapter($qb));
+        $pager->setMaxPerPage($maxPerPage);
+        $pager->setCurrentPage($page);
+
+        return $pager;
+    }
+
+    public function paginate(ListOptions $options, QueryBuilder $queryBuilder = null): Pagerfanta
+    {
+        if (null === $queryBuilder) {
+            $queryBuilder = $this->getAllQb();
+        }
+
+        try {
+            $page = $options->get('page');
+            $limit = $options->get('limit');
+
+            if (!\is_int($page) || !\is_int($limit)) {
+                throw new \RuntimeException('int parameter');
+            }
+        } catch (\RuntimeException $e) {
+            $page = 1;
+            $limit = 10;
+        }
+
+        return $this->getPager(
+            $queryBuilder,
+            $page,
+            $limit,
+        );
+    }
+
+    public function getFilteredResource(
+        FormInterface $form,
+        ?array $sorting = [],
+        QueryBuilder $qb = null,
+    ): QueryBuilder {
+        $queryBuilder = $qb ?? $this->getCollectionQueryBuilder();
+
+        $this->applyFilters($form, $queryBuilder);
+
+        if (!empty($sorting)) {
+            $this->applySorting($queryBuilder, $sorting);
+        }
+
+        return $queryBuilder;
+    }
+
+    protected function getPropertyName(string $name): string
+    {
+        if (!str_contains($name, '.')) {
+            return $this->getAlias().'.'.$name;
+        }
+
+        return $name;
+    }
+
+    protected function applySorting(QueryBuilder $queryBuilder, array $sorting = []): void
+    {
+        foreach ($sorting as $property => $order) {
+            if (!empty($order)) {
+                $queryBuilder->addOrderBy($this->getPropertyName($property), $order);
+            }
+        }
+    }
+
+    protected function getAlias(): string
+    {
+        return 'o';
+    }
+
+    protected function applyFilters(FormInterface $form, QueryBuilder $queryBuilder): void
+    {
+        if (null === $form->getData()) {
+            return;
+        }
+        
+        $this->filterBuilderUpdater->addFilterConditions($form, $queryBuilder);
+    }
+```
+
+Bonus: Twig list component
+==========================
+
+```php
+#[AsTwigComponent(name: 'crud-list', template: 'components/crud/list.html.twig')]
```

Comment:
Yea, that looks pretty interesting - I "get" your use-case. It could be a nice demo on ux.symfony.com, though right now, we only have demos for "live" components. I'd like to create a more general "cookbook" section of examples, but it doesn't exist yet.

But, we could still add it to the "live components" demo section for now with a note - then move it later.